lib/storage.js

```diff
@@ -0,0 +1,111 @@
+'use strict';
+
+const UPDATE_TIMEOUT_MS = 500;
+
+/**
+ * Check to if the given value is mergable.
+ *
+ * @param {*} value
+ *
+ * @return {Boolean}  Returns true if the object can be merged, false otherwise.
+ */
+function canMerge(value) {
+  return value !== null && typeof value === 'object' && !Array.isArray(value);
+}
+
+/**
+ * Check if the given value will overwrite other values during a merge.
+ *
+ * @param {*} value
+ *
+ * @return {Boolean} Returns true if the value will overwrite others, false otherwise.
+ */
+function canOverwriteWith(value) {
+  return value !== null && value !== undefined && !(value instanceof Function);
+}
+
+/**
+ * Perform a deep merge of objects based on key.
+ * Warning! Destination is modified in place.
```

Comment:
I haven't found a merge library that does what I want yet. For the object case, I want a deep merge based on keys. For _everything else_ I want to overwrite existing values. This mostly breaks downf or `Array`. Everything I've found seems to want to do

```javascript
merge(['a', 'b'], ['c']) === ['c', 'b'])
```